R/bundlePackage.R

```diff
@@ -74,63 +57,15 @@ checkBundlePackages <- function(deps, call = caller_env()) {
   }
 }
 
-## check for extra dependencies uses
-inferRPackageDependencies <- function(appMode,
-                                      hasParameters = FALSE,
-                                      documentsHavePython = FALSE) {
-  deps <- switch(appMode,
-    "rmd-static" = c("rmarkdown", if (hasParameters) "shiny"),
-    "quarto-static" = "rmarkdown",
-    "quarto-shiny" = c("rmarkdown", "shiny"),
-    "rmd-shiny" = c("rmarkdown", "shiny"),
-    "shiny" = "shiny",
-    "api" = "plumber"
-  )
-  if (documentsHavePython) {
-    deps <- c(deps, "reticulate")
-  }
-  deps
-}
-
-appUsesR <- function(quartoInfo) {
-  if (is.null(quartoInfo)) {
-    # All non-Quarto content currently uses R by default.
-    # To support non-R content in rsconnect, we could inspect appmode here.
-    return(TRUE)
-  }
-  # R is used only supported with the "knitr" engine, not "jupyter" or "markdown"
-  # Technically, "jupyter" content could support R.
-  return("knitr" %in% quartoInfo[["engines"]])
-}
-
-
-preservePackageDescriptions <- function(bundleDir) {
-  # Copy all the DESCRIPTION files we're relying on into packrat/desc.
-  # That directory will contain one file for each package, e.g.
-  # packrat/desc/shiny will be the shiny package's DESCRIPTION.
-  #
-  # The server will use this to calculate package hashes. We don't want
-  # to rely on hashes calculated by our version of packrat, because the
-  # server may be running a different version.
-  lockFilePath <- snapshotLockFile(bundleDir)
+# Copy all the DESCRIPTION files we're relying on into packrat/desc.
+# That directory will contain one file for each package, e.g.
+# packrat/desc/shiny will be the shiny package's DESCRIPTION.
+copyPackageDescriptions <- function(bundleDir, packages) {
   descDir <- file.path(bundleDir, "packrat", "desc")
-  tryCatch({
-    dir.create(descDir)
-    records <- utils::tail(read.dcf(lockFilePath), -1)
-    lapply(seq_len(nrow(records)), function(i) {
-      pkgName <- records[i, "Package"]
-      descFile <- system.file("DESCRIPTION", package = pkgName)
-      if (!file.exists(descFile)) {
-        stop("Couldn't find DESCRIPTION file for ", pkgName)
-      }
-      file.copy(descFile, file.path(descDir, pkgName))
-    })
-  }, error = function(e) {
-    warning("Unable to package DESCRIPTION files: ", conditionMessage(e), call. = FALSE)
-    if (dirExists(descDir)) {
-      unlink(descDir, recursive = TRUE)
-    }
-  })
+  dir.create(descDir, showWarnings = FALSE, recursive = TRUE)
+
+  descPaths <- file.path(find.package(packages), "DESCRIPTION")
```

Comment:
The try/catch and file-existence checks feel like they could help identify malformed libraries, but I don't know that those cases have ever been tripped.

The previous pattern was in place in the initial implementation, rather than reacting to reported issues:
https://github.com/rstudio/rsconnect/pull/42

Reply:
`checkBundlePackages()` catches the most common case of the package not being installed and if your library is sufficiently messed up that the `DESCRIPTION` isn't installed, I suspect you'll get an upstream error (i.e. packrat will also have problems).